scalameta/src/main/scala/scala/meta/Trees.scala

```diff
@@ -170,7 +175,7 @@ package scala.meta.internal.ast {
     @ast class While(expr: Term, body: Term) extends Term
     @ast class Do(body: Term, expr: Term) extends Term
     @ast class For(enums: Seq[Enumerator] @nonEmpty, body: Term) extends Term with Scope {
-      require(enums.head.isInstanceOf[Enumerator.Generator])
+      require(enums.head.isInstanceOf[Enumerator.Generator] || enums.head.isInstanceOf[Enumerator.Generator.Quasi])
```

Comment:
Do we really need this? Isn't `Enumerator.Generator.Quasi` an `Enumerator.Generator`?